src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -599,5 +583,88 @@ private static string GetReferencedModel(String schema, Dictionary<string, Schem
 
         public static IEnumerable<string> GetParentTrackedResources(IEnumerable<string> trackedResourceModels, IEnumerable<KeyValuePair<string, string>> childTrackedResourceModels)
             => trackedResourceModels.Where(resourceModel => !childTrackedResourceModels.Any(childModel => childModel.Key.Equals(resourceModel)));
+
+        /// <summary>
+        /// For the provided resource model, it gets the operation which ends with ListByResourceGroup and returns the resource model.
+        /// </summary>
+        /// <param name="resourceModel"></param>
+        /// <param name="definitions"></param>
+        /// <param name="serviceDefinition"></param>
+        /// <returns></returns>
+        public static Operation GetListByResourceGroupOperation(string resourceModel, Dictionary<string, Schema> definitions, ServiceDefinition serviceDefinition)
+        {
+            return GetListByXOperation(resourceModel, definitions, serviceDefinition, listByRgRegEx);
+        }
+
+        /// <summary>
+        /// For the provided resource model, it gets the operation which matches with ListBySubscription and returns the resource model.
+        /// </summary>
+        /// <param name="resourceModel"></param>
+        /// <param name="definitions"></param>
+        /// <param name="serviceDefinition"></param>
+        /// <returns></returns>
+        public static Operation GetListBySubscriptionOperation(string resourceModel, Dictionary<string, Schema> definitions, ServiceDefinition serviceDefinition)
+        {
+            return GetListByXOperation(resourceModel, definitions, serviceDefinition, listBySidRegEx);
+        }
+
+        /// <summary>
+        /// For the provided resource model, it gets the operation which matches with specified regex and returns the resource model.
+        /// </summary>
+        /// <param name="resourceModel"></param>
+        /// <param name="definitions"></param>
+        /// <param name="serviceDefinition"></param>
+        /// <param name="regEx"></param>
+        /// <returns></returns>
+        private static Operation GetListByXOperation(string resourceModel, Dictionary<string, Schema> definitions, ServiceDefinition serviceDefinition, Regex regEx)
+        {
+            return GetListByOperation(regEx, resourceModel, definitions, serviceDefinition);
+        }
+
+        /// <summary>
+        /// Checks if there is an operation that matches the given regex
+        /// </summary>
+        /// <param name="getOperations"></param>
+        /// <param name="regEx"></param>
+        /// <param name="definitionKey"></param>
+        /// <param name="definitions"></param>
+        /// <returns></returns>
+        public static bool ListByXCheck(Regex regEx, string definitionKey, Dictionary<string, Schema> definitions, ServiceDefinition serviceDefinition)
```

Comment:
is this method needed? looks like it's a small wrapper for GetListByOperation, though returning boolean. Should we have a more representative name?